Examples/restcomm-olympus/app/src/androidTest/java/org/restcomm/android/olympus/IntegrationTests.java

```diff
@@ -1029,6 +1046,511 @@ public void run() {
+                params.put(RCDevice.ParameterKeys.DEBUG_JAIN_DISABLE_CERTIFICATE_VERIFICATION, true);
+
+                params.put(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_APPLICATION_NAME, PUSH_APPLICATION_NAME);
+                params.put(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_ACCOUNT_EMAIL, PUSH_ACCOUNT);
+                params.put(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_ACCOUNT_PASSWORD, PUSH_PASSWORD);
+                params.put(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_PUSH_DOMAIN, PUSH_DOMAIN);
+                params.put(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_HTTP_DOMAIN, HTTP_DOMAIN);
+                params.put(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_FCM_SERVER_KEY, PUSH_FCM_KEY);
+
+
+                device.setLogLevel(Log.VERBOSE);
+
+                try {
+                    device.initialize(InstrumentationRegistry.getTargetContext(), params, IntegrationTests.this);
+                } catch (RCException e) {
+                    Log.e(TAG, "RCDevice Initialization Error: " + e.errorText);
+                }
+            }
+        });
+        await().atMost(PUSH_TIMEOUT, TimeUnit.SECONDS).until(fieldIn(this).ofType(boolean.class).andWithName("onWarningReceived"), equalTo(true));
+        assertThat(context.get("status-code")).isEqualTo(RCClient.ErrorCodes.SUCCESS.ordinal());
+
+        onWarningReceived = false;
+
+        //disable
+        params.put(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_ENABLE_PUSH_FOR_ACCOUNT, false);
+        clientHandler.post(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    device.updateParams(params);
+                } catch (Exception e) {
+                    Log.e(TAG, "RCDevice update Error: " + e.toString());
+                }
+            }
+        });
+
+        await().atMost(PUSH_TIMEOUT, TimeUnit.SECONDS).until(fieldIn(this).ofType(boolean.class).andWithName("onWarningReceived"), equalTo(true));
+        assertThat(context.get("status-code")).isEqualTo(RCClient.ErrorCodes.SUCCESS.ordinal());
+
+        //if the binding is missing its removed.
+        StorageManagerPreferences storageManagerPreferences = new StorageManagerPreferences(InstrumentationRegistry.getTargetContext());
+        assertThat(TextUtils.isEmpty(storageManagerPreferences.getString(FcmConfigurationHandler.FCM_BINDING,"")));
+        clientHandler.post(new Runnable() {
+            @Override
+            public void run() {
+                resetPushConfiguration();
+                device.release();
+            }
+        });
+
+        await().atMost(SIGNALING_TIMEOUT, TimeUnit.SECONDS).until(fieldIn(this).ofType(boolean.class).andWithName("deviceReleased"), equalTo(true));
+        InstrumentationRegistry.getTargetContext().unbindService(this);
+    }
+
+    @Test(timeout = TC_TIMEOUT)
+    public void deviceUpdateParamsWithPushEnabled() throws InterruptedException {
```

Comment:
I can't find a call to updateParams() in this TC. Am i missing sth?

Reply:
I think initialization will do it, but you are right. Update should be done.